Library/Homebrew/build.rb

```diff
@@ -172,8 +172,7 @@ def fixopt f
 end
 
 begin
-  error_pipe = IO.new(ENV["HOMEBREW_ERROR_PIPE"].to_i, "w")
-  error_pipe.fcntl(Fcntl::F_SETFD, Fcntl::FD_CLOEXEC)
+  error_pipe = UNIXSocket.open(ENV["HOMEBREW_ERROR_PIPE"]).recv_io
```

Comment:
Thinking about this code, I don't know why we are even setting this up prior to installation. It's only used in failure cases. Can we just do something like this in the rescue block (I don't know, I haven't tested it)?

```rb
UnixSocket.open(ENV["HOMEBREW_ERROR_PIPE"]) { |s| Marshal.dump(e, s.recv_io) }
```